processor-api/src/main/java/com/synopsys/integration/alert/processor/api/NotificationProcessorV2.java

```diff
@@ -69,13 +69,13 @@ private void processAndDistribute(List<AlertNotificationModel> notifications, Co
                                                                         .map(notificationDetailExtractionDelegator::wrapNotification)
                                                                         .flatMap(List::stream)
                                                                         .collect(Collectors.toList());
-        StatefulAlertPagedModel<FilteredJobNotificationWrapper> mappedNotifications = jobNotificationMapper.mapJobsToNotifications(filterableNotifications, frequencies);
+        StatefulAlertPage<FilteredJobNotificationWrapper> mappedNotifications = jobNotificationMapper.mapJobsToNotifications(filterableNotifications, frequencies);
         do {
-            for (FilteredJobNotificationWrapper jobNotificationWrapper : mappedNotifications.getCurrentModels()) {
+            for (FilteredJobNotificationWrapper jobNotificationWrapper : mappedNotifications.getCurrentPage().getModels()) {
                 processAndDistribute(jobNotificationWrapper);
             }
             mappedNotifications = mappedNotifications.retrieveNextPage();
-        } while (mappedNotifications.hasNextPage());
+        } while (mappedNotifications.hasNextPage() || !mappedNotifications.isEmpty());
```

Comment:
Mentioned this earlier but the logic here will be somewhat changed after I implement the JobNotificationMapper correctly in the future.